app.py

```diff
@@ -70,6 +71,44 @@
 
 app = Flask(__name__, static_url_path='')
 
+def into_number (string):
+    total = 0
+    for char in string:
+        total += ord (char)
+    return total
+
+def redirect_ab (request, session):
+    # If the user is logged in, we use their username as identifier, otherwise we use the session id
+    user_identifier = current_user(request) ['username'] or str (session_id)
+
+    # This will send 50% of the requests into redirect.
+    redirect_flag = (into_number (user_identifier) % 100) < 50
```

Comment:
The % of how many users you want to test should probably be a parameter. Also, 50% seems extremely high.